homeassistant/components/huisbaasje/sensor.py

```diff
@@ -0,0 +1,255 @@
+
+    # Create a coordinator for polling updates
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="sensor",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=POLLING_INTERVAL),
+    )
+
+    # Fetch initial data
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        [
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity Rate",
+                device_class=DEVICE_CLASS_POWER,
+                source_type=SOURCE_TYPE_ELECTRICITY,
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity In Rate",
+                device_class=DEVICE_CLASS_POWER,
+                source_type=SOURCE_TYPE_ELECTRICITY_IN,
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity In Low Rate",
+                device_class=DEVICE_CLASS_POWER,
+                source_type=SOURCE_TYPE_ELECTRICITY_IN_LOW,
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity Out Rate",
+                device_class=DEVICE_CLASS_POWER,
+                source_type=SOURCE_TYPE_ELECTRICITY_OUT,
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity Out Low Rate",
+                device_class=DEVICE_CLASS_POWER,
+                source_type=SOURCE_TYPE_ELECTRICITY_OUT_LOW,
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity Today",
+                unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+                source_type=SOURCE_TYPE_ELECTRICITY,
+                sensor_type=SENSOR_TYPE_THIS_DAY,
+                icon="mdi:counter",
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Gas Rate",
+                unit_of_measurement=FLOW_CUBIC_METERS_PER_HOUR,
+                source_type=SOURCE_TYPE_GAS,
+                icon="mdi:fire",
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Gas Today",
+                unit_of_measurement=VOLUME_CUBIC_METERS,
+                source_type=SOURCE_TYPE_GAS,
+                sensor_type=SENSOR_TYPE_THIS_DAY,
+                icon="mdi:counter",
+            ),
+        ]
+    )
+
+
+def _get_measurement_rate(current_measurements: dict, source_type: str):
+    if source_type in current_measurements.keys():
+        if current_measurements[source_type]["measurement"]:
+            return current_measurements[source_type]["measurement"]["rate"]
+    else:
+        _LOGGER.warn(
```

Comment:
Please fix these pylint warnings. E.g.: `_LOGGER.warning("Source type %s not present in %s",source_type, current_measurements)`